server.js

```diff
@@ -0,0 +1,18 @@
+require("dotenv").config();
+
+const { createApp } = require("./app");
+const { appDataSource } = require("./models/data-source");
+
+const startServer = async () => {
+  const app = createApp();
+  const PORT = process.env.PORT;
+
+  await appDataSource.initialize();
+  console.log("database has been initialized!");
+
+  app.listen(PORT, () => {
+    console.log(`Listening on Port ${PORT}`);
+  });
+};
```

Comment:
typeorm의 dataSource를 활용하여 데이터베이스와 연동하는 작업을 하는 경우 비동기적으로 처리하는 코드를 잘 작성해주셨습니다.

다만 데이터베이스에 제대로 연동이 되지 않았을 경우, 로그를 확인할 수 있는 에러 핸들링을 작성해주면 좋을 것 같습니다.

아래 코드를 참고하여 수정해주세요!
혹은 [공식문서](https://orkhan.gitbook.io/typeorm/docs/data-source#how-to-use-datasource)를 참고하여 수정해주세요!
```suggestion
const startServer = async () => {
  try {
    const app = createApp();
    const PORT = process.env.PORT;

    await appDataSource.initialize();
    console.log("database has been initialized!");

    app.listen(PORT, () => {
      console.log(`Listening on Port ${PORT}`);
    });
  } catch(err) {
    console.log("Error during Data Source initialization", err);
  }
};
```